python-package/xgboost/core.py

```diff
@@ -1240,6 +1251,70 @@ def get_quantile_cut(self) -> Tuple[np.ndarray, np.ndarray]:
         assert data.dtype == np.float32
         return indptr, data
 
+    def get_categories(self) -> Optional[Dict[str, "pa.DictionaryArray"]]:
```

Comment:
We might one day generalize this into an `arrow encoder/decoder`. For now, it's just glue code.